service/entityresolution/keycloak/keycloak_entity_resolution.go

```diff
@@ -28,12 +30,18 @@ const UsernameConditionalSelector = "client_id"
 const serviceAccountUsernamePrefix = "service-account-"
 
 type KeycloakConfig struct {
-	URL            string `json:"url"`
-	Realm          string `json:"realm"`
-	ClientID       string `json:"clientid"`
-	ClientSecret   string `json:"clientsecret"`
-	LegacyKeycloak bool   `json:"legacykeycloak" default:"false"`
-	SubGroups      bool   `json:"subgroups" default:"false"`
+	URL                 string              `json:"url"`
+	Realm               string              `json:"realm"`
+	ClientID            string              `json:"clientid"`
+	ClientSecret        string              `json:"clientsecret"`
+	LegacyKeycloak      bool                `json:"legacykeycloak" default:"false"`
+	SubGroups           bool                `json:"subgroups" default:"false"`
+	UnknownEntityConfig UnknownEntityConfig `json:"unknownentities" default:"false"`
+}
+
+type UnknownEntityConfig struct {
+	Enabled     bool     `json:"enabled"`
```

Comment:
Instead of a type here, you can have a method that checks if EntityTypes is empty